File: Samples/State Pattern/Sample_1 - CS/AggressiveState.cs
namespace Sample
{
    internal class AggressiveState : State
    {
        public override void Enter()
        {
            Console.WriteLine($"[{_enemy.GetName}.AggressiveState]: called Enter!");
        }

        public override void Exit()
        {
            Console.WriteLine($"[{_enemy.GetName}.AggressiveState]: called Exit!");
        }

        public override void Update()
        {
            Console.WriteLine($"[{_enemy.GetName}.AggressiveState]: called Update!");
        }
    }
}

File: Samples/State Pattern/Sample_1 - CPP/AggressiveState.cpp
#include "AggressiveState.h"

void AggressiveState::Enter()
{
	cout << "[" << m_enemy->GetName() << ".AggressiveState] called Enter!" << endl;
}

void AggressiveState::Update()
{
	cout << "[" << m_enemy->GetName() << ".AggressiveState] called Update!" << endl;
}

void AggressiveState::Exit()
{
	cout << "[" << m_enemy->GetName() << ".AggressiveState] called Exit!" << endl;
}
